python/whylogs/core/dataset_profile.py

```diff
@@ -140,13 +140,11 @@ def _do_track(
             self._initialize_new_columns(tuple(new_cols))
 
         if row is not None:
-            if col_id is not None:
-                logger.warning(
-                    "validation with identity column is not supported for row level tracking.\
-                    Validation will be performed without identity column"
-                )
             for k in row.keys():
-                self._columns[k]._track_datum(row[k])
+                if col_id:
```

Comment:
could do `for key, value in row.items():` so you could replace `row[k]` with `value` in the loop body